### DISCUSSION TITLE
ESP reset

### DISCUSSION THREAD

**Discussion by RobertBarbo** [Dlloydev/QuickPID#72] [General]
Hello,

Am using QuickPID library i must say it works good, for heating purposes. But i have one problem. Sometimes my ESP32 resets and i get masive undershoot, couse all PID parameters are set to default (calculated). And it takes time to PID starts controlling. 
Is there a way to save those calculated values and restore them at reboot?
![Brez naslova](https://github.com/Dlloydev/QuickPID/assets/78436393/df4dd11f-afc9-4ee0-a1de-5cc7f64903fd)

Thank you for your help.

**Comment by Dlloydev** [comment]
You could look at [using the preferences library](https://randomnerdtutorials.com/esp32-save-data-permanently-preferences/), but I think it would be better to identify and resolve the resetting issue.

Perhaps its EMI/RFI from switching the heater load with a relay or an SSR. Depending on your circuit, you may only need to add an MOV or snubber, provide further separation from the AC circuit and wiring to the low voltage DC and MCU board.

It could also be some issue with your code.

**Comment by RobertBarbo** [comment]
OK, thank you for your reply. Can you tell me what parameters for PID are crutial to save at reboot? Am thinking the one from Reset(): outputSum = 0;
  pTerm = 0;
  iTerm = 0;
  dTerm = 0;

**Comment by Dlloydev** [comment]
The only measurement based parameter that could have significant effect would be outputSum. Other parameters that are based on each sample wouldn't have any significant value. An unknown portion of the dropout is probably unavoidable, as the heater would loose power until after reboot and previous settings are re established.

Currently, there's no method to set the outputSum, but I'll add this shortly in a minor update and report back when its ready.
EDIT: QuickPID 3.1.9 published (includes the SetOutputSum() function).
